ruma-identity-service-api/src/lookup/lookup_3pid/v2.rs

```diff
@@ -0,0 +1,52 @@
+//! [POST /_matrix/identity/v2/lookup](https://matrix.org/docs/spec/identity_service/latest#post-matrix-identity-v2-lookup)
+
+use std::collections::BTreeMap;
+
+use ruma_api::ruma_api;
+
+ruma_api! {
+    metadata: {
+        description: "Looks up the set of Matrix User IDs which have bound the 3PIDs given, if bindings are available.",
+        method: POST,
+        name: "lookup_3pid",
+        path: "/_matrix/identity/v2/lookup",
+        authentication: AccessToken,
+        rate_limited: false,
+    }
+
+    request: {
+        /// The algorithm the client is using to encode the `addresses`. This should be one of the
+        /// available options from `/hash_details`.
+        pub algorithm: &'a str,
+
+        /// The pepper from `/hash_details`. This is required even when the `algorithm` does not
+        /// make use of it.
+        pub pepper: &'a str,
+
+        /// The addresses to look up. The format of the entries here depend on the `algorithm`
+        /// used. Note that queries which have been incorrectly hashed or formatted will lead to no
+        /// matches.
+        pub addresses: Vec<String>
+    }
+
+    response: {
+        /// Any applicable mappings of `addresses` to Matrix User IDs. Addresses which do not have
+        /// associations will not be included, which can make this property be an empty object.
+        pub mappings: BTreeMap<String, String>,
```

Comment:
We use the identifier types for common identifiers in the spec. (See [here](https://docs.ruma.io/ruma_identifiers/index.html) for a current list.)

```suggestion
        pub mappings: BTreeMap<String, UserId>,
```